### PR TITLE
[Enhancement] Support query iceberg tables with binary columns

### PR DESCRIPTION
**Pull request by letian-jiang**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

* Convert binary type in iceberg to varbinary type in starrocks
* Skip conversion because iceberg binary types are length-variable.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by letian-jiang** [comment]
run starrocks_clang-format 
run starrocks_admit_test

**Comment by letian-jiang** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/connector/ColumnTypeConverter.java [hunk 1]
```diff
@@ -436,6 +436,7 @@ public static Type fromIcebergType(org.apache.iceberg.types.Type icebergType) {
             case TIME:
             case FIXED:
             case BINARY:
+                return Type.VARBINARY;
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/ColumnTypeConverter.java:439]: convert TIME, FIXED,BINARY to VARBINARY?

Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/connector/ColumnTypeConverter.java:437]: fixed

**Comment by Smith-Cruise** [comment]
How about orc, we support orc already?

**Review by Youngwb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|0|1|00.00%|[437]|

**Comment by stephen-shelby** [comment]
test other type tables with varbinary column

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|0|1|00.00%|[437]|

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
Have supported ORC by adding fill_functions for binary types

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|0|1|00.00%|[437]|

File: be/src/formats/orc/fill_function.cpp [hunk 2]
```diff
@@ -702,6 +702,75 @@ static void fill_string_column_with_null(orc::ColumnVectorBatch* cvb, ColumnPtr&
...
+
+    int col_start = col->size();
+
+    auto* c = ColumnHelper::as_raw_column<NullableColumn>(col);
+
+    c->null_column()->resize(col->size() + size);
+    auto* nulls = c->null_column()->get_data().data();
+    auto* values = ColumnHelper::cast_to_raw<TYPE_VARBINARY>(c->data_column());
+
+    values->get_offset().reserve(values->get_offset().size() + size);
```
Comment (Smith-Cruise) [inline be/src/formats/orc/fill_function.cpp:747]: You can write `values->get_offset().reserve(col_start + size);` instead, keep the same style of `fill_varbinary_column`, and also you can change `fill_string_column_with_null` ethier.

**Review by Smith-Cruise — COMMENTED** [review]
Plz add orc `orc_chunk_reader_test.cpp` and parquet `file_reader_test.cpp` UT respectively.

File: be/src/formats/orc/fill_function.cpp [hunk 3]
```diff
@@ -702,6 +702,75 @@ static void fill_string_column_with_null(orc::ColumnVectorBatch* cvb, ColumnPtr&
...
+    size_t len = 0;
+    for (size_t i = 0; i < size; ++i) {
+        len += data->length[from + i];
+    }
+
+    int col_start = col->size();
+    auto* values = ColumnHelper::cast_to_raw<TYPE_VARBINARY>(col);
+
+    values->get_offset().reserve(col_start + size);
+    values->get_bytes().reserve(len);
```
[the remark this answers is not in the document]
Comment (letian-jiang) [reply be/src/formats/orc/fill_function.cpp]: done

File: be/src/formats/orc/fill_function.cpp (hunk 2 above)
Comment (letian-jiang) [reply be/src/formats/orc/fill_function.cpp]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|0|1|00.00%|[437]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|0|1|00.00%|[437]|

**Comment by letian-jiang** [comment]
run starrocks_admit_test

File: be/src/formats/orc/fill_function.cpp [hunk 4]
```diff
@@ -702,6 +702,75 @@ static void fill_string_column_with_null(orc::ColumnVectorBatch* cvb, ColumnPtr&
...
+
+    for (int i = col_start; i < col_start + size; ++i, ++pos) {
+        vb.insert(vb.end(), data->data[pos], data->data[pos] + data->length[pos]);
+        vo.emplace_back(vb.size());
+    }
+}
+
+static void fill_varbinary_column_with_null(orc::ColumnVectorBatch* cvb, ColumnPtr& col, size_t from, size_t size,
+                                            const TypeDescriptor& type_desc, const OrcMappingPtr& mapping, void* ctx) {
+    auto* data = down_cast<orc::StringVectorBatch*>(cvb);
```
Comment (stephen-shelby) [inline be/src/formats/orc/fill_function.cpp:732]: only support String?

Comment (letian-jiang) [reply be/src/formats/orc/fill_function.cpp:732]: orc string and binary use the same type of `ColumnVectorBatch`

https://github.com/letian-jiang/starrocks/blob/0617986c708fcb06cafdcfee1a27089c189b1640/be/src/formats/orc/apache-orc/c++/src/TypeImpl.cc#L327
https://github.com/StarRocks/starrocks/blob/0617986c708fcb06cafdcfee1a27089c189b1640/be/src/formats/orc/apache-orc/c++/src/TypeImpl.cc#L327

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|0|1|00.00%|[437]|

**Comment by letian-jiang** [comment]
run starrocks_be_unittest

**Review by Smith-Cruise — DISMISSED** [review]

**Comment by DorianZheng** [comment]
lack of parquet UT

**Review by DorianZheng — DISMISSED** [review]
Please add UT for parquet reader. rest LGTM.

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
run starrocks_fe_unittest

**Comment by letian-jiang** [comment]
run starrocks_fe_unittest

**Review by Smith-Cruise — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|0|1|00.00%|[437]|

**Review by stephen-shelby — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Youngwb — DISMISSED** [review]

**Review by Smith-Cruise — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by letian-jiang** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**